osu.Game/Rulesets/Mods/ModRateAdjust.cs

```diff
@@ -2,12 +2,13 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using osu.Framework.Audio;
+using osu.Framework.Audio.Sample;
 using osu.Framework.Audio.Track;
 using osu.Framework.Bindables;
 
 namespace osu.Game.Rulesets.Mods
 {
-    public abstract class ModRateAdjust : Mod, IApplicableToTrack
+    public abstract class ModRateAdjust : Mod, IApplicableToTrack, IApplicableToSample
```

Comment:
This is a minor point, but I find the double interface specification a little iffy/fine-grained. It's not unreasonable that someone might find `IAppplicableToTrack` but not find/forget about `IApplicableToSample` (as you even did in 4138f61).

I don't think the two should be merged outright, but how about a grouping interface?

```csharp
public interface IApplicableToAudio : IApplicableToTrack, IApplicableToSample
{
}
```